### ISSUE TITLE
FNX-12158 ⁃ [Bug]Tabs Tray: Crash: java.util.ConcurrentModificationException: at java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java)

### ISSUE THREAD

**Issue by AndiAJ** [mozilla-mobile/fenix#11260]
## Prerequisites
Have more then 10 open tabs

## Steps to reproduce
1. Open one of the listed tabs
1. Tap the tab counter
1. Open another tab
1. Put Fenix to background
1. ResumeFenix
1. Open the Tabs Tray
1. Open a tab

### Expected behavior
Properly load the open tabs

### Actual behavior
Crash

### Device information
* Android device: 
• Google Pixel 3a (Android 10)
• OnePlus A3 (Android 6.0.1)
* Fenix version: Nightly Build 200605 from 6/5

### Notes
```
java.util.ConcurrentModificationException
* Sentry: https://sentry.prod.mozaws.net/operations/fenix-nightly/?query=66b16d667e32454eaae509857dee0a11
----
java.util.ConcurrentModificationException
at java.util.LinkedHashMap$LinkedHashIterator.nextEntry(LinkedHashMap.java:346)
at java.util.LinkedHashMap$KeyIterator.next(LinkedHashMap.java:366)
at mozilla.components.support.base.observer.ObserverRegistry.notifyObservers(ObserverRegistry.kt:2)
at mozilla.components.browser.tabstray.TabsAdapter.notifyObservers(TabsAdapter.kt)
at org.mozilla.fenix.tabtray.TabTrayViewHolder$bind$3.onClick(TabTrayViewHolder.kt:1)
at android.view.View.performClick(View.java:5233)
at android.view.View$PerformClick.run(View.java:21209)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:152)
at android.app.ActivityThread.main(ActivityThread.java:5507)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)

```

Crash report from Google Pixel 3a : https://crash-stats.mozilla.org/report/index/991f1e5b-87c3-4181-88c5-072bb0200605

► Video
![20200605-152448](https://user-images.githubusercontent.com/51314259/83876681-eaaadb00-a741-11ea-9193-3f48b47b0063.gif)

Related to the issue from #11174

**Comment by ekager** [comment]
I can reproduce 100% of the time with just 1 tab open.

**Comment by jonalmeida** [comment]
Sentry issue: [FENIX-NIGHTLY-38J](https://sentry.prod.mozaws.net/operations/fenix-nightly/issues/8639659/?referrer=github_plugin)

**Comment by AndiAJ** [comment]
Hi @ekager & @boek , I can't properly test this matter on the latest **`Nightly Build 200609 from 6/9`** due to the fact that **none of the open tabs can be opened on first attempt after resuming Fenix**.
Submitted this new issue #11361 

I'll leave the QA needed label on this ticket.

**Comment by AndiAJ** [comment]
Hi @ekager I've just re-checked this on both the latest **`Nightly Build 20610 from 6/10`** & **`Beta 5.2.0-beta.1 from 6/10`** using the following devices:
• Google Pixel 3a (Android 10)
• OnePlus A3 (Android 6.0.1)

## Beta
#11361 **is still reproducible**

## Nightly
❗  **Native code crash on both devices**

• Google Pixel 3a (Android 10) [Crash report](https://crash-stats.mozilla.org/report/index/a5cae651-103f-48d0-9b66-159c80200610) 


• OnePlus A3 (Android 6.0.1)

► Log
```kotlin
06-10 13:30:57.311 5409-5422/? E/mozac/CrashReporter: CrashHandlerService received native code crash
06-10 13:30:57.314 5409-5422/? I/mozac/CrashReporter: Received crash: NativeCodeCrash(minidumpPath=/data/user/0/org.mozilla.fenix.nightly/files/mozilla/6n0hnf5v.default/minidumps/7620fcb5-8531-a515-4937-9c32d6dc6a3b.dmp, minidumpSuccess=false, extrasPath=/data/user/0/org.mozilla.fenix.nightly/files/mozilla/6n0hnf5v.default/minidumps/7620fcb5-8531-a515-4937-9c32d6dc6a3b.extra, isFatal=true, breadcrumbs=[], uuid=e1951d50-cf6d-4f58-b3c6-26faa3180a60)
06-10 13:30:57.367 5409-5422/? I/mozac/CrashReporter: Showing prompt
06-10 13:30:57.368 1302-2189/? I/ActivityManager: START u0 {flg=0x10018000 cmp=org.mozilla.fenix.nightly/mozilla.components.lib.crash.prompt.CrashReporterActivity (has extras)} from uid 10277 pid 5409 on display 0
06-10 13:30:57.375 4821-4973/? I/Gecko: Exiting due to channel error.
06-10 13:30:57.400 1302-2189/? E/JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 112)
06-10 13:30:57.401 1302-2189/? W/ActivityManager: Exception thrown during pause
    android.os.DeadObjectException: Transaction failed on small parcel; remote process probably died
        at android.os.BinderProxy.transactNative(Native Method)
        at android.os.BinderProxy.transact(Binder.java:527)
        at android.app.ApplicationThreadProxy.schedulePauseActivity(ApplicationThreadNative.java:734)
        at com.android.server.am.ActivityStack.startPausingLocked(ActivityStack.java:887)
        at com.android.server.am.ActivityStack.finishActivityLocked(ActivityStack.java:2970)
        at com.android.server.am.TaskRecord.performClearTaskAtIndexLocked(TaskRecord.java:653)
        at com.android.server.am.TaskRecord.performClearTaskLocked(TaskRecord.java:666)
        at com.android.server.am.ActivityStackSupervisor.startActivityUncheckedLocked(ActivityStackSupervisor.java:2220)
        at com.android.server.am.ActivityStackSupervisor.startActivityLocked(ActivityStackSupervisor.java:1743)
        at com.android.server.am.ActivityStackSupervisor.startActivityMayWait(ActivityStackSupervisor.java:1084)
        at com.android.server.am.ActivityManagerService.startActivityAsUser(ActivityManagerService.java:4285)
        at com.android.server.am.ActivityManagerService.startActivity(ActivityManagerService.java:4247)
        at android.app.ActivityManagerNative.onTransact(ActivityManagerNative.java:162)
        at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2762)
        at android.os.Binder.execTransact(Binder.java:458)
06-10 13:30:57.415 720-720/? I/Zygote: Process 4741 exited cleanly (6)
06-10 13:30:57.416 1302-1365/? E/JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 60)
06-10 13:30:57.423 2171-2171/? D/Launcher: onResume
06-10 13:30:57.425 2171-2171/? D/AppTracker: App Event: start
06-10 13:30:57.432 2171-2171/? D/Launcher: onPause
06-10 13:30:57.432 2171-2171/? D/AppTracker: App Event: stop
06-10 13:30:57.446 1302-2250/? V/BoostFramework: BoostFramework() : mPerf = com.qualcomm.qti.Performance@4bcdb9
06-10 13:30:57.446 1302-2250/? I/ActivityManager: Start proc 5424:org.mozilla.fenix.nightly:mozilla.components.lib.crash.CrashReporter/u0a277 for activity org.mozilla.fenix.nightly/mozilla.components.lib.crash.prompt.CrashReporterActivity
06-10 13:30:57.452 5409-5422/? I/art: System.exit called, status: 0
06-10 13:30:57.453 5409-5422/? I/AndroidRuntime: VM exiting with result code 0, cleanup skipped.
06-10 13:30:57.474 1302-1302/? I/DeviceIdleController: updateChargingLocked: charging=true
06-10 13:30:57.474 1302-1302/? I/DeviceIdleController: becomeActiveLocked, reason = charging
06-10 13:30:57.495 1302-1326/? I/ActivityManager: Process org.mozilla.fenix.nightly:mozilla.components.lib.crash.CrashHandler (pid 5409) has died
```

► Video
![20200610-133044](https://user-images.githubusercontent.com/51314259/84260242-629e4a00-ab22-11ea-8aff-d75db4cd61a8.gif)

Please review and share your thoughts. ☺️ 
I'll remove the QA needed label

**Comment by kbrosnan** [comment]
@AndiAJ that looks to be completely different. Please report a new issue. This will be fixed on beta with the 5.2 release.

BZ-1644043

https://crash-stats.mozilla.org/signature/?product=Fenix&signature=java.util.ConcurrentModificationException%3A%20at%20java.util.LinkedHashMap%24LinkedHashIterator.nextNode%28LinkedHashMap.java%29
